lib/membrane_opus/parser/frame_lengths.ex

```diff
@@ -0,0 +1,192 @@
+defmodule Membrane.Opus.Parser.FrameLengths do
+  @moduledoc false
+  # Helper for Membrane.Opus.Parser for determining frame lengths
+
+  # given a frame length, return opus binary encoded length
+  @spec encode_frame_length(non_neg_integer) :: binary
+  def encode_frame_length(length) do
+    if length < 252 do
+      <<length::size(8)>>
+    else
+      <<252 + rem(length - 252, 4)::size(8), div(length - 252, 4)::size(8)>>
+    end
+  end
+
+  # returns ordered list of frame lengths and header length
+  @spec parse_frame_lengths(0..3, binary, boolean) :: {[non_neg_integer], pos_integer}
```

Comment:
don't i need the size really though, in order to do delimit/undelimit? and since i'm already pulling that seems easier. but are you saying that returning the header itself would be more _readable_, as opposed to more efficient? If more readable, then I'll make the change.

Reply:
I don't think there will be much difference regarding performance, since we have really small chunks here, moreover, if you just extract parts of a binary, VM just passes references to the original binary. Regarding readability, I think it's more readable when you don't need to add one byte for header there and two bytes elsewhere. Anyway, I don't think it will be easy to refactor that, so maybe just give it a try.